qiskit/algorithms/amplitude_amplifiers/amplitude_amplifier.py

```diff
@@ -43,14 +44,14 @@ class AmplitudeAmplifierResult(AlgorithmResult):
 
     def __init__(self) -> None:
         super().__init__()
-        self._top_measurement = None
+        self._top_measurement: str | None = None
         self._assignment = None
-        self._oracle_evaluation = None
-        self._circuit_results = None
-        self._max_probability = None
+        self._oracle_evaluation: bool | None = None
+        self._circuit_results: list[np.ndarray] | list[dict[str, int]] | None = None
```

Comment:
This annotation is duplicated in the return of `circuit_results` method. You should make a type alias `CircuitResultsT = list[np.ndarray] | list[dict[str, int]] | None`. And I think you have to use `Optional` and `Union` for the type alias declaration. Because it is not supported yet until Python 3.10.

Reply:
Actually, you should exclude the ` | None` in the `CircuitResultsT` declaration, because the `circuit_results` setter doesn't expect `None`.